app/gui2/src/components/GraphEditor.vue

```diff
@@ -255,11 +257,20 @@ const graphBindingsHandler = graphBindings.handler({
   },
   collapse() {
     if (keyboardBusy()) return false
-    const selected = nodeSelection.selected
+    const selected = new Set(nodeSelection.selected)
     if (selected.size == 0) return
     try {
-      const info = prepareCollapsedInfo(nodeSelection.selected, graphStore.db)
-      performCollapse(info)
+      const info = prepareCollapsedInfo(selected, graphStore.db)
+      const currentMethod = projectStore.executionContext.getStackTop()
+      const currentMethodName = graphStore.db.stackItemToMethodName(currentMethod)
+      if (currentMethodName == null) {
+        bail(`Cannot get the method name for the current execution stack item. ${currentMethod}`)
+      }
+      graphStore.editAst((module) => {
+        if (graphStore.moduleRoot == null) bail(`Module root is missing.`)
+        const topLevel = module.get(graphStore.moduleRoot)! as Ast.BodyBlock
```

Comment:
Instead of a type cast and `!`, this should be an `assert`.

```suggestion
        const topLevel = module.get(graphStore.moduleRoot)
        assert(topLevel instanceof Ast.BodyBlock)
```

Reply:
Done